packages/asana/src/Utils.js

```diff
@@ -0,0 +1,44 @@
+import { composeNextState } from '@openfn/language-common';
+import {
+  request as commonRequest,
+  logResponse,
+} from '@openfn/language-common/util';
+
+export function addAuth(headers, configuration = {}) {
+  const { token } = configuration;
+  if (token) {
+    Object.assign(headers, { Authorization: `Bearer ${token}` });
+  }
+}
+
+export function request(state, path, params, callback = s => s) {
+  let { body, headers = {}, method = 'GET', ...rest } = params;
+
+  const baseUrl = `https://app.asana.com/api/${state.configuration?.apiVersion}`;
+
+  addAuth(headers, state.configuration);
+
+  const options = {
+    ...rest,
+    headers,
+    baseUrl,
+    body,
+  };
+
+  return commonRequest(method, path, options)
+    .then(response => {
+      logResponse(response);
+
+      return {
+        ...composeNextState(state, response.body?.data),
+        response,
```

Comment:
Something I've just done in `openmrs` is to remove the `body` from the response object,  because otherwise it just spams up state (the response data will be duplicated on `state.response` and `state.data`). 

Admittedly it's a drop in the ocean compared to references, but it does bother me a bit. 

Should we do the same here? 

See here: https://github.com/OpenFn/adaptors/blob/904e434cbb5f3e388715a2cbac2564700f732ea2/packages/openmrs/src/Utils.js#L9